README.md

```diff
@@ -1,2 +1,307 @@
+image, or some partial image that used to do something small. In this
+example we pull in 2 different base images. One is a named layer for
+the golang build environment, so that it can be referenced later. The
+second one is the base image with just enough operating system to
+provide debug facilities inside the image.
+
+RUN - executes a command, all output files go into the layer. There
+are a few `go get` commands that pull in libraries for the main
+application. Later the main.go is compiled using this.
+
+COPY - copy a file from the local filesystem into the container. This
+can work on a set of files.
+
+WORKDIR - set the working directory for the entrypoint
+
+CMD - what command should be run in this container if no exec is
+passed into it. CMD means that the command run is replaceable. There
+is another stanza `ENTRYPOINT` which is not replaceable.
+
+EXPOSE - open up a specific port from this container to the
+infrastructure. Without this no network traffic will be allowed into
+the container.
+
+These commands, plus many more are documented on the [Docker reference
+site](https://docs.docker.com/engine/reference/builder/#cmd).
+
+## Best Practices for Building Images
+
+There are a number of best practices when building images.
+
+### Clean up everything you can
+
+Whatever files are created in a layer are stored in the intermediate
+layer. Sometimes what gets added in each layer ends up as a
+surprise. For instance, when install Linux packages with `apt`, not
+only is the software installed, but copies of the packages are stored
+in `/var/cache/apt/archives`. This can add significant bloat to your
+images, which slows down image upload/download times and thus
+container start times.
+
+When using standard install tools, be sure to include clean up
+commands as part of the run command. For instance:
+
+```
+RUN apt-get update && apt-get install -y python3 python3-pip && apt-get clean
+```
+
+Which updates the repos, installs packages, and deletes downloaded
+archive files.
+
+# Publishing to IBM Cloud
+
+You can build images with docker and publish them publicly to docker
+hub. However, if you are going to be using these images inside of IBM
+cloud it's best practice to publish them into the IBM Cloud Container
+Registry. This makes the images much more network local to where they
+run, and makes them private to your IBM Cloud account.
+
+## Creating your registry
+
+Before you can upload images to the IBM Cloud you must first create a
+registry namespace. Namespaces are just used to help you organize your
+account. You can have many namespaces in your account, creating one
+per project is a useful way of staying organized.
+
+```
+> bx cr namespace-add guestbook
+```
+
+This will create a registry url of `registry.ng.bluemix.net/guestbook`
+that you can use for uploading images.
+
+## Building using bx client
+
+The bx client can be used to build images using the local docker
```

Comment:
Is this true? I thought using 'bx' would build in IBM Cloud, not local. To do local build don't they need to use "docker build"?

Reply:
I am pretty sure it's a local build

Reply:
Can't tell for sure from the output but either way, it does push to the registry so the net result isn't local. But do a "docker images" and see if you see the image

Reply:
Ok, cleared up now